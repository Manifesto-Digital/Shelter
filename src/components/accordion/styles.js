import styled from 'styled-components';
import SVG from 'react-inlinesvg';
import { buttonReset } from '../styled/buttons';

export const Wrapper = styled.div`
  position: relative;

  &:not(:first-of-type) {
    &:before {
      position: absolute;
      content: ' ';
      left: 0;
      top: ${({ theme }) => `-${theme.spacing.small}`};
      width: 100%;
      border-top: ${({ theme, active }) =>
        active && `1px solid ${theme.palette.grey45}`};
    }
  }
`;

export const Heading = styled.h3`
  position: relative;
  font-weight: normal;
  color: ${({ theme }) => theme.palette.grey10};
  margin-bottom: ${({ active }) => active && '0'};
`;

export const HeadingButton = styled.button`
  position: relative;
  ${buttonReset};
  padding: ${({ theme }) =>
    `${theme.spacing.standard} ${theme.spacing.xl} ${theme.spacing.standard} ${
      theme.spacing.standard
    }`};
  width: 100%;
  text-align: left;
  font-size: ${({ theme }) => theme.headers.h3};
`;

export const Content = styled.div`
  display: ${({ isOpen }) => (isOpen ? 'block' : 'none')};
  padding: ${({ theme }) =>
    `${theme.spacing.standard} ${theme.spacing.standard} 0 ${
      theme.spacing.standard
    }`};
`;

export const ArrowSVG = styled(SVG)`
  position: absolute;
  transform: ${({ isOpen }) => isOpen && 'rotate(180deg)'} translateY(-50%);
  transform-origin: top;
  top: 50%;
  right: ${({ theme }) => theme.spacing.standard};
  width: 20px;
  height: 25px;

  svg {
    display: block;
    width: 20px;
    height: 25px;
  }
`;
