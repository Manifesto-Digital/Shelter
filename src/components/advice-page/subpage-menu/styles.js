import styled from 'styled-components';

export const MenuNav = styled.nav`
  margin-bottom: ${({ theme }) => theme.spacing.large};

  a {
    text-decoration: none;
  }
`;
