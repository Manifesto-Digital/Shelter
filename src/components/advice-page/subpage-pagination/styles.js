import styled from 'styled-components';
import SVG from 'react-inlinesvg';

export const StyledLink = styled.a`
  text-decoration: none;
  margin-right: ${({ type }) => (type === 'prev' ? 'auto' : null)};
  margin-left: ${({ type }) => (type === 'next' ? 'auto' : null)};

  span {
    text-decoration: underline;
    font-weight: normal;
  }
`;

export const DirectionIcon = styled.div`
  display: flex;
  flex-direction: ${({ type }) => (type === 'prev' ? 'row-reverse' : 'row')};
  justify-content: flex-end;
  font-size: ${({ theme }) => theme.fontsize.larger};
  margin: 0 0 0.4em;
`;

export const Wrapper = styled.div`
  display: flex;
  margin-bottom: ${({ theme }) => theme.spacing.large};
  padding-bottom: ${({ theme }) => theme.spacing.large};
  border-bottom: 2px solid ${({ theme }) => theme.palette.offWhite};
`;

export const StyledSVG = styled(SVG)`
  svg {
    display: block;
    width: 35px;
    height: 40px;
  }
`;
