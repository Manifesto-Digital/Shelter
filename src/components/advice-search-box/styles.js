import styled from 'styled-components';
import Button from '../button';
import { breakpoint } from '../theme/breakpoint';
import TextInput from '../forms/text-input';

export const AdviceSearchForm = styled.form`
  padding: ${props => props.theme.spacing.standard};
  background-color: ${props => props.theme.palette.offWhite};
`;

export const SearchInput = styled(TextInput)`
  margin-bottom: 1em;
  ${breakpoint.tablet`
        display: inline-block;
        width: 75%;
        margin-right: 2em;
    `};
`;

export const SearchButton = styled(Button)`
  background: ${props => props.theme.palette.sanMarinoBlue};
  position: relative;
  width: auto;
  padding: ${props => props.theme.spacing.small};
  color: ${props => props.theme.palette.white};
  display: block;
  ${breakpoint.tablet`
        display: inline-block;
    `};
`;
