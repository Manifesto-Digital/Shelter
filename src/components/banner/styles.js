import styled from 'styled-components'
import LinkHandler from '../link-handler'

export const Wrapper = styled.div`
  padding: ${props => props.theme.spacing.padding} 0;
  background-color: ${props =>
    (props.bg === 'red' && props.theme.palette.primary) ||
    (props.bg === 'blue' && props.theme.palette.sanMarino) ||
    (props.bg === 'black' && props.theme.palette.black) ||
    (props.bg === 'green' && props.theme.palette.donate)};
  margin-bottom: ${props =>
    props.removeMarginBottom ? '0' : props.theme.spacing.largePadding};
`

export const Header = styled.h2`
  color: ${props =>
    (props.bg === 'red' && props.theme.palette.white) ||
    (props.bg === 'blue' && props.theme.palette.white) ||
    (props.bg === 'black' && props.theme.palette.white) ||
    (props.bg === 'green' && props.theme.palette.white)};
`
export const StyledLinkHandler = styled(LinkHandler)`
  color: ${props =>
    (props.bg === 'red' && props.theme.palette.white) ||
    (props.bg === 'blue' && props.theme.palette.white) ||
    (props.bg === 'black' && props.theme.palette.white) ||
    (props.bg === 'green' && props.theme.palette.white)};
`
