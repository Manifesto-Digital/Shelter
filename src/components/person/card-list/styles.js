import styled, { css } from 'styled-components';
import { buttonReset } from '../../styled/buttons';
import { breakpoint } from '../../theme/breakpoint';

export const Wrapper = styled.section`
  margin-bottom: ${({ theme }) => theme.spacing.large};
  ${({ borderTop }) =>
    borderTop &&
    css`
      margin-top: ${({ theme }) => theme.spacing.large};
      border-top: 1px solid ${({ theme }) => theme.palette.grey10};
      padding-top: ${({ theme }) => theme.spacing.standard};
    `};
`;

export const List = styled.ul`
  padding: 0;
  display: grid;

  ${breakpoint.desktop`
    grid-template-columns: ${({ columns }) =>
      columns && `repeat(${columns}, 1fr)`};
    grid-gap: ${({ theme }) => theme.spacing.standard};
  `};
`;

export const LoadMore = styled.div`
  text-align: center;
  visibility: ${({ visible }) => (visible ? 'visible' : 'hidden')};
`;

export const LoadMoreButton = styled.button`
  ${buttonReset}
  color: ${({ theme }) => theme.palette.primary};
  padding: ${({ theme }) => theme.spacing.small};
  
  svg { 
    width: 35px;
    height: 35px;
  }
`;
