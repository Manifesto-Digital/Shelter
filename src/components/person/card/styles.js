import SVG from 'react-inlinesvg';
import styled from 'styled-components';
import { breakpoint } from '../../theme/breakpoint';
import Image from '../../image';

export const Card = styled.div`
  display: flex;
  flex-wrap: wrap;
  box-shadow: ${({ theme }) => theme.boxshadow.small};
  position: relative;
  margin-bottom: ${({ theme }) => theme.spacing.standard};
  width: 100%;

  &:hover {
    box-shadow: ${({ hasLink, theme }) => hasLink && theme.boxshadow.standard};
  }

  ${breakpoint.desktop`
    width: ${({ theme, columns }) =>
      columns === 2 && `calc(50% - ${theme.spacing.small})`};

    font-size: ${({ theme, columns }) => columns && theme.fontsize.small};
  `};
`;

export const Info = styled.div`
  padding: ${({ theme }) => `${theme.spacing.small} ${theme.spacing.standard}`};
  flex: 1;
`;

export const Name = styled.h3`
  margin-bottom: 5px;
`;

export const JobTitle = styled.p`
  margin-bottom: ${({ theme }) => theme.spacing.small};
  font-weight: ${({ theme }) => theme.fontWeight.bold};
  line-height: 1.2;
`;

export const StyledImage = styled(Image)`
  align-self: center;
  margin-bottom: ${({ theme }) => theme.spacing.small};
  width: 100%;

  ${breakpoint.mobileLand`
    width: 35%;
    margin-bottom: 0;
  `};
`;

export const ContactLine = styled.p`
  margin-bottom: 5px;

  &:last-of-type {
    margin-bottom: 0;
  }
`;

export const ContactIcon = styled(SVG)`
  margin-right: ${({ theme }) => theme.spacing.small};

  svg {
    width: 18px;
    height: 18px;
  }
  path {
    fill: ${({ theme }) => theme.palette.primary};
  }
`;
