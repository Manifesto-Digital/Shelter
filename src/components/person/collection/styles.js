import styled from 'styled-components';

export const Wrapper = styled.section`
  margin-bottom: ${({ theme }) => theme.spacing.large};
  padding-top: ${({ theme }) => theme.spacing.standard};
`;

export const Header = styled.h2`
  margin-bottom: ${({ theme }) => theme.spacing.large};
`;
