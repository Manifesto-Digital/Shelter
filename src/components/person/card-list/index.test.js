import React from 'react';
import { shallow } from 'enzyme';
import {
  snapshotComponent,
  mountWithTheme,
} from '../../../../__tests__/helpers';
import { createFactory, createPerson } from '../../../utils/test-factories';
import CardList from '.';
import PersonCard from '../card';
import { List, LoadMoreButton } from './styles';
import { sizes, emSize } from '../../theme/breakpoint';

const createCardListProps = createFactory({
  limit: 9,
  list: [
    createPerson({
      id: 'id1',
      firstName: 'mock first name',
      lastName: 'mock last name',
    }),
    createPerson({
      id: 'id2',
      firstName: 'mock first name 2',
      lastName: 'mock last name 2',
    }),
  ],
});

it('renders correctly', () => {
  const mockData = createCardListProps();
  snapshotComponent(<CardList {...mockData} />);
});

it('renders a person for each item', () => {
  const mockData = createCardListProps();
  const wrapper = shallow(<CardList {...mockData} />);
  expect(wrapper.find(PersonCard).length).toEqual(mockData.list.length);
});

it('sets column width depending on prop', () => {
  const mockData = createCardListProps({
    columns: 2,
  });
  const wrapper = mountWithTheme(<CardList {...mockData} />);
  expect(wrapper.find(List)).toHaveStyleRule(
    'grid-template-columns',
    expect.stringContaining(mockData.columns.toString()), // Allow for any gutter size in calc
    {
      media: `(min-width: ${emSize(sizes.desktop)})`,
    }
  );
});

it('will show more person cards if possible', () => {
  const mockData = createCardListProps({
    limit: 1,
  });
  const wrapper = mountWithTheme(<CardList {...mockData} />);

  expect(wrapper.find(PersonCard).length).toEqual(mockData.limit);
  // More should have been loaded
  wrapper.find(LoadMoreButton).simulate('click');
  expect(wrapper.find(PersonCard).length).toBeGreaterThan(mockData.limit);
});
