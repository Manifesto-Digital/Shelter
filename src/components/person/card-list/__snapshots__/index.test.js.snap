// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
.c1 {
  max-width: 1140px;
  margin: 0 auto;
  padding: 0 20px;
}

.c10 {
  border-bottom-right-radius: 3px;
  background: #8c8c8c;
  color: #fff;
  padding-top: 3px;
  height: 30px;
  width: 30px;
  position: absolute;
  right: 3px;
  bottom: 3px;
}

.c10 svg {
  width: 22px;
  height: 22px;
  display: block;
  margin: 0 auto;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  box-shadow: 0 1.5px 4px rgba(0,0,0,0.12),0 1.5px 6px rgba(0,0,0,0.06);
  position: relative;
  margin-bottom: 20px;
  width: 100%;
}

.c5 {
  padding: 10px 20px;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.c6 {
  margin-bottom: 5px;
}

.c7 {
  margin-bottom: 10px;
  font-weight: 500;
  line-height: 1.2;
}

.c4 {
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  margin-bottom: 10px;
  width: 100%;
}

.c8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  word-break: break-word;
  margin-bottom: 10px;
  line-height: 1;
}

.c8:last-of-type {
  margin-bottom: 0;
}

.c9 {
  margin-right: 10px;
}

.c9 svg {
  width: 18px;
  height: 18px;
}

.c9 path {
  fill: #ea232d;
}

.c0 {
  margin-bottom: 40px;
}

.c2 {
  padding: 0;
  display: grid;
}

@media (min-width:62em) {
  .c3 {
    margin-bottom: 0;
    font-size: 0.775rem;
  }
}

@media (min-width:32em) {
  .c5 {
    width: 70%;
    margin-bottom: 0;
  }
}

@media (min-width:32em) {
  .c4 {
    -webkit-align-self: flex-start;
    -ms-flex-item-align: start;
    align-self: flex-start;
    margin-top: 10px;
    width: 30%;
    margin-bottom: 0;
  }
}

@media (min-width:62em) {
  .c2 {
    grid-template-columns: repeat(2,1fr);
    grid-gap: 20px;
  }
}

<section
  className="c0"
>
  <div
    className="c1"
  >
    <ul
      className="c2"
    >
      <li
        className="c3"
      >
        <img
          alt="Collection hero"
          className="c4"
          src="//images.ctfassets.net/6sxvmndnpn0s/2DPKnmx9Na8WYgG4ySqkA/5ca89770eb1ac36c9dbfe34d8d65eb5c/collection-hero.jpg?fit=fill&h=500&w=500"
        />
        <div
          className="c5"
        >
          <h3
            className="c6"
          >
            mock first name
             
            mock last name
          </h3>
          <p
            className="c7"
          >
            Developer
          </p>
          <p
            className="c8"
          >
            <span
              className="isvg loading c9"
            />
             
            test@test.com
          </p>
          <p
            className="c8"
          >
            <span
              className="isvg loading c9"
            />
             
            012345678910
          </p>
        </div>
        <div
          aria-hidden="true"
          className="c10"
        >
          <span
            className="isvg loading"
          />
        </div>
      </li>
      <li
        className="c3"
      >
        <img
          alt="Collection hero"
          className="c4"
          src="//images.ctfassets.net/6sxvmndnpn0s/2DPKnmx9Na8WYgG4ySqkA/5ca89770eb1ac36c9dbfe34d8d65eb5c/collection-hero.jpg?fit=fill&h=500&w=500"
        />
        <div
          className="c5"
        >
          <h3
            className="c6"
          >
            mock first name 2
             
            mock last name 2
          </h3>
          <p
            className="c7"
          >
            Developer
          </p>
          <p
            className="c8"
          >
            <span
              className="isvg loading c9"
            />
             
            test@test.com
          </p>
          <p
            className="c8"
          >
            <span
              className="isvg loading c9"
            />
             
            012345678910
          </p>
        </div>
        <div
          aria-hidden="true"
          className="c10"
        >
          <span
            className="isvg loading"
          />
        </div>
      </li>
    </ul>
  </div>
</section>
`;
